src/components/injector.js

```diff
@@ -66,7 +67,7 @@ export function injector(map = {}) {
         return injector(map);
       }
 
-      return injector({...map, [factory]: replacement});
+      return injector((new Map(map)).set(factory, replacement));
```

Comment:
why do you need to create a new Map here? isn't `map` already a `Map`?

Reply:
This keeps maps immutable, which is good IMO.